### DISCUSSION TITLE
customize theme

### DISCUSSION THREAD

**Discussion by skyward06** [ant-design/ant-design#51116] [Q&A]
How can I customize the default theme of Ant Design components?

**Comment by petercat-assistant** [comment]
To customize the default theme of Ant Design components, you can follow these steps:

1. **Using Less Variables**: Ant Design allows you to customize the theme by overriding Less variables. You can find the list of all available variables in the [Ant Design documentation](https://ant.design/docs/react/customize-theme). You can modify these variables in your project's Less file to change the appearance of the components.

2. **Using `@ant-design/cssinjs`**: This package provides a way to customize the theme using JavaScript. You can define your theme variables in a JavaScript object and apply them globally.

3. **Using `antd`'s `ConfigProvider`**: You can use the `ConfigProvider` component to set global configurations, including theme customization. This is useful for setting themes dynamically at runtime.

4. **Customizing Specific Components**: If you want to customize specific components, you can use styled-components or CSS modules to apply styles directly to the components.

5. **Webpack Configuration**: If you are using a build tool like Webpack, you can configure it to modify the Less variables during the build process. This involves using `less-loader` to pass your custom variables.

Here's a simple example of how you might customize the theme using Less variables:

```less
// variables.less
@primary-color: #1DA57A; // Change the primary color
@link-color: #1DA57A; // Change the link color
```

Then, import this Less file into your project to apply the changes.

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Comment by Stormer0528** [comment]
To customize the default theme in Ant Design, you can use Less variables. First, install Less and the necessary loader if you're using a bundler like Webpack. Then, you can create a Less file to override the default variables. For example:

```
@primary-color: #1DA57A; // Change the primary color
@link-color: #1DA57A; // Change the link color

@import '~antd/dist/antd.less'; // Import Ant Design styles

```
